src/api/api.c

```diff
@@ -746,15 +747,15 @@ int8_t z_publisher_put(const z_publisher_t pub, const uint8_t *payload, size_t l
                    opt.attachment
 #endif
     );
-
+#if Z_FEATURE_SUBSCRIPTION == 1
```

Comment:
You shouldn't need this because `_z_trigger_local_subscriptions` is replaced by a dummy function if subscription function is off.... Suposedly

Reply:
see comment above, it doesn´t compile. Feels like code was tested with this feature off.

Reply:
No it's definitely tested with both in CI, I recall you use C++ to compile, so maybe this is the source of our compilation diverging behavior.